internal/compression/deflate.go

```diff
@@ -31,33 +33,21 @@ func (c *deflateDecompressor) Read(bytes []byte) (int, error) {
 	return c.reader.Read(bytes)
 }
 func (c *deflateDecompressor) Reset(rdr io.Reader) error {
-	resetter, ok := c.reader.(flate.Resetter)
-	if !ok {
-		// This should never happen as the returned type from flate should always
-		// implement Resetter, but the check is here as a safeguard just in case.
-		// This error would be a very exceptional / unexpected occurrence.
-		return errors.New("deflate reader is not able to be used as a resetter")
-	}
-	// Mimics NewReader internal logic, which initializes the internal dict to nil
-	return resetter.Reset(rdr, nil)
+	var err error
+	c.reader, err = zlib.NewReader(rdr)
+	return err
 }
 func (c *deflateDecompressor) Close() error {
 	return c.reader.Close()
 }
 
 // NewDeflateDecompressor returns a new deflate Decompressor.
 func NewDeflateDecompressor() connect.Decompressor {
-	return &deflateDecompressor{
-		reader: flate.NewReader(nil),
-	}
+	return &deflateDecompressor{}
```

Comment:
Ditto -- could either initialize `reader` field to something like `errorDecompressor{err: io.EOF}` or could add appropriate nil guards in other methods, to avoid panics.

Reply:
I added nil checks in the other methods, too. That should cover our bases pretty well I think.